src/v/cluster/cluster_discovery.cc

```diff
@@ -69,22 +71,34 @@ ss::future<node_id> cluster_discovery::determine_node_id() {
     co_return assigned_node_id;
 }
 
-std::vector<model::broker> cluster_discovery::initial_seed_brokers() const {
+ss::future<cluster_discovery::brokers>
+cluster_discovery::initial_seed_brokers() {
     // If configured as the root node, we'll want to start the cluster with
     // just this node as the initial seed.
     if (config::node().empty_seed_starts_cluster()) {
         if (config::node().seed_servers().empty()) {
-            return {make_self_broker(config::node())};
+            co_return brokers{make_self_broker(config::node())};
         }
         // Not a root
-        return {};
+        co_return brokers{};
     }
     if (get_node_index_in_seed_servers()) {
-        // TODO: return the discovered nodes plus this node
-        return {make_self_broker(config::node())};
+        co_await fetch_cluster_bootstrap_info();
+        brokers seed_brokers = get_seed_brokers_for_bootstrap();
+        seed_brokers.push_back(make_self_broker(config::node()));
+        co_return std::move(seed_brokers);
     }
     // Non-seed server
-    return {};
+    co_return brokers{};
+}
+
+cluster_discovery::brokers
+cluster_discovery::initial_seed_brokers_if_no_cluster(
+  const std::optional<model::cluster_uuid>& stored_cluster_uuid) {
+    if (stored_cluster_uuid.has_value()) {
+        return {};
+    }
```

Comment:
nit: rather than forcing the orchestration code to pass cluster uuid around, it seems like it would simplify things to have the stored cluster UUID as a member of `cluster_discovery`.